dask/array/_array_expr/_collection.py

```diff
@@ -97,6 +111,46 @@ def size(self) -> T_IntOrNaN:
     def name(self):
         return self.expr.name
 
+    def __len__(self):
+        return self.expr.__len__()
+
+    def __getitem__(self, index):
+        # Field access, e.g. x['a'] or x[['a', 'b']]
+        if isinstance(index, str) or (
+            isinstance(index, list) and index and all(isinstance(i, str) for i in index)
+        ):
+            # TODO(expr-soon): needs map_blocks that we don't support yet,
+            #  but implementation is trivial after we have that
+            raise NotImplementedError()
```

Comment:
below you're adding map_blocks. What else is missing?

Reply:
nothing, but I forgot about this. The implementation is part of the ufunc branch which comes after this one